launcher/common/cosimulator.py

```diff
@@ -0,0 +1,201 @@
+        self.__action_plan_dict = self.__xml_manager_plan.get_action_plan_dict()
+
+        ########
+        # STEP 5 Co-Simulation Parameters
+        ########
+        self.__xml_manager_parameters = \
+            common.xml_manager_parameters.XmlManagerParameters(configuration_manager=self.__configuration_manager,
+                                                               logger=self.__logger,
+                                                               xml_filename=self.__args.parameters)
+
+        # STEP 5.1 - Dissecting Co-Simulation Parameters XML file
+        if not self.__xml_manager_parameters.dissect() == common.enums.XmlManagerReturnCodes.XML_OK:
+            return common.enums.CoSimulatorReturnCodes.XML_ERROR
+
+        # STEP 5.2 - Getting the variables found in the Co-Simulation Parameters file
+        self.__parameters_variables_dict = self.__xml_manager_parameters.get_variables_dict()
+
+        # STEP 5.3 - Getting the parameters found in the Co-Simulation Parameters file
+        self.__parameters_parameters_dict = self.__xml_manager_parameters.get_parameters_dict()
+
+        ########
+        # STEP 6 - Co-Simulation Actions (processing the XML configuration files)
+        ########
+        # STEP 6.1 - Getting the Actions Popen arguments, the CO_SIM_ variables transformation is performed 
+        self.__xml_manager_actions = \
+            common.xml_manager_actions.XmlManagerActions(configuration_manager=self.__configuration_manager,
+                                                         logger=self.__logger,
+                                                         environment_variables_manager=self.__variables_manager,
+                                                         action_plan=self.__action_plan_dict)
+        if not self.__xml_manager_actions.dissect() == common.enums.XmlManagerReturnCodes.XML_OK:
+            return common.enums.CoSimulatorReturnCodes.XML_ERROR
+
+        self.__actions_popen_args_dict = self.__xml_manager_actions.get_actions_popen_arguments_dict()
+        
+        ########
+        # STEP 7 - Converting Co-Simulation XML parameters into JSON
+        ########
+        if not self.generate_parameters_json_file() == common.enums.CoSimulatorReturnCodes.OK:
+            return common.enums.CoSimulatorReturnCodes.JSON_FILE_ERROR
+
+        ########
+        # STEP 8 - Launching action plan
+        ########
+        self.__launcher = common.Launcher(action_plan_dict=self.__action_plan_dict,
+                                          actions_popen_args_dict=self.__actions_popen_args_dict,
+                                          configuration_manager=self.__configuration_manager,
+                                          logger=self.__logger)
+        if not self.__launcher.carry_out_action_plan() == common.enums.LauncherReturnCodes.LAUNCHER_OK:
+            return common.enums.CoSimulatorReturnCodes.LAUNCHER_ERROR
+
+        # STEP 99 - Finishing
```

Comment:
Good point!, So far it's 99 just because I didn't know how many of them will be added, at the moment when he have a steady version and we now more/less the numbers of "steps" I will changes to the proper correlative one.